extension/content.js

```diff
@@ -881,13 +882,14 @@
       if (errorMsg.indexOf("GUM") === 0) {
         errorMsg =  "Please enable your microphone to use Voice Fill";
       } else if (errorMsg.indexOf("EMPTYRESULTS") === 0) {
-        errorMsg = "No results found";  
+        errorMsg = "No results found";
       } else {
         errorMsg = "Sorry, we encountered an error";
       }
       loadAnimation(ERROR_ANIMATION, false);
       const copy = document.getElementById("stm-content");
-      copy.innerHTML = `<div id="stm-listening-text">${errorMsg}</div>`
+      const error = DOMPurify.sanitize(errorMsg);
```

Comment:
It seems that the input is text. `DOMPurify` assumes that the input is HTML, and if the input contains HTML, the result is still HTML (sanitized, but unescaped).

When you have text input and want to convert that to something that can safely be used as HTML, then escape significant characters, e.g. as follows: https://gist.github.com/Rob--W/ec23b9d6db9e56b7e4563f1544e0d546

Reply:
Thanks @Rob--W good points!

@andrenatal If we only insert text and don't need to inject HTML, then we can switch the places we're using DOMPurify to escape their input instead and use DOM APIs like [textContent](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent) instead of innerHTML. Let me know if that's the case and I'll make an issue and try to get around to patching it sooner than later.